File: LeetCode/871.keys-and-rooms/keys-and-rooms.java
class Solution {
    public boolean canVisitAllRooms(List<List<Integer>> rooms) {
        Stack<Integer>stk = new Stack<>();
        stk.add(0);
        boolean[] vis = new boolean[rooms.size()];
        Arrays.fill(vis,false);
        vis[0] = true;
        while (!stk.empty()){
            int index = stk.pop();
            for (int val : rooms.get(index)){
                if (!vis[val]){
                    stk.add(val);
                    vis[val] = true;
                }
            }
        }
        for (boolean val : vis){
            if (!val){
                return false;
            }
        }
        return true;
    }
}

File: LeetCode/0841.keys-and-rooms/keys-and-rooms.cpp
class Solution {
public:
    bool canVisitAllRooms(vector<vector<int>>& rooms) {
        bool vis[1005];
        memset(vis,false,sizeof(vis));
        queue<int>que;
        que.push(0);
        vis[0] = true;
        while (!que.empty()){
        	int index = que.front();
            que.pop();
        	for (int val : rooms[index]){
        		if (!vis[val]){
        			que.push(val);
                    
        			vis[val] = true;
        		}
        	}
        }
        for (int i = 0;i < rooms.size();i++){
        	if (!vis[i])    return false;
        }
        return true;
    }
};